src/overlays/actors/ovl_En_Mm3/z_en_mm3.c

```diff
@@ -381,18 +382,20 @@ void func_80A6FBA0(EnMm3* this) {
 void func_80A6FBFC(EnMm3* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
-    if (gSaveContext.unk_3DD0[0] == 0x10) {
+    if (gSaveContext.timerStates[TIMER_ID_POSTMAN] == TIMER_STATE_POSTMAN_END) {
         player->stateFlags1 &= ~0x20;
         this->actor.flags |= ACTOR_FLAG_10000;
-        if (gSaveContext.unk_3DE0[0] > 1500) {
-            gSaveContext.unk_3DE0[0] = 1500;
-        } else if ((((void)0, gSaveContext.unk_3DE0[0]) >= (OSTime)(995 - XREG(16))) &&
-                   (((void)0, gSaveContext.unk_3DE0[0]) <= (OSTime)(1005 + XREG(17)))) {
-            gSaveContext.unk_3DE0[0] = 1000;
+        if (gSaveContext.timerCurTimes[TIMER_ID_POSTMAN] > SECONDS_TO_TIMER(15)) {
+            gSaveContext.timerCurTimes[TIMER_ID_POSTMAN] = SECONDS_TO_TIMER(15);
+        } else if ((((void)0, gSaveContext.timerCurTimes[TIMER_ID_POSTMAN]) >=
+                    (OSTime)(SECONDS_TO_TIMER_PRECISE(10, -5) - XREG(16))) &&
+                   (((void)0, gSaveContext.timerCurTimes[TIMER_ID_POSTMAN]) <=
+                    (OSTime)(SECONDS_TO_TIMER_PRECISE(10, 5) + XREG(17)))) {
+            gSaveContext.timerCurTimes[TIMER_ID_POSTMAN] = SECONDS_TO_TIMER(10);
         }
-    } else if (gSaveContext.unk_3DE0[0] > 1500) {
-        gSaveContext.unk_3DD0[0] = 15;
-        gSaveContext.unk_3DC8 = osGetTime();
+    } else if (gSaveContext.timerCurTimes[TIMER_ID_POSTMAN] > SECONDS_TO_TIMER(15)) {
+        gSaveContext.timerStates[TIMER_ID_POSTMAN] = TIMER_STATE_POSTMAN_STOP;
+        gSaveContext.postmanTimerStopOsTime = osGetTime();
```

Comment:
We are going to have to look into that VC thing for JP at some point.

Reply:
I'm also quite curious about the postman's timer differences in JP